src/CryptographicHash.h

```diff
@@ -0,0 +1,64 @@
+// Copyright 2005-2017 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#ifndef MUMBLE_CRYPTOGRAPHICHASH_H_
+#define MUMBLE_CRYPTOGRAPHICHASH_H_
+
+#include <QByteArray>
+
+class CryptographicHashPrivate;
+
+/// CryptographicHash is used for computing cryptographic hashes.
+class CryptographicHash {
+	public:
+		/// Algorithm specifies an algorithm for use with CryptographicHash.
+		enum Algorithm {
+			/// The SHA-1 hashing algorithm.
+			Sha1,
+			/// The SHA-256 hashing algorithm.
+			Sha256
+		};
+
+		/// Consruct a CryptographicHash object that can be used
+		/// to incrementally compute a cryptographic hash using the
+		/// hash function specified in |algo|.
+		CryptographicHash(Algorithm algo);
+
+		/// addData adds the content of |buf| to the hash computation.
+		void addData(const QByteArray &buf);
+
+		/// result() finalizes the CryptographicHash object,
+		/// and returns the computed hash.
+		/// If an error occurred during the computatiopn, the
+		/// returned QByteArray will be empty.
+		QByteArray result() const;
+
+		/// Compute the cryptographic hash of |buf| using algorithm |algo|
```

Comment:
It's a Chromium/BoringSSL style thing. I quite like it, because it lets you write a full comment for humans, instead of having to repeat yourself with `@params`, etc.

I don't have a strong preference here, though. `@params` is obviously a bit nicer with IDEs that show parameter info on hover.

Kept it for now.